server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionNode.java

```diff
@@ -0,0 +1,440 @@
+    return picClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition has occurs clause
+   *
+   * @return true if the variable definition has occurs clause
+   */
+  public boolean hasOccurs() {
+    return !occursClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition doesn't have occurs clause
+   *
+   * @return true if the variable definition doesn't have occurs clause
+   */
+  public boolean doesntHaveOccurs() {
+    return occursClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition has usage clause
+   *
+   * @return true if the variable definition has usage clause
+   */
+  public boolean hasUsage() {
+    return !usageClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition doesn't have usage clause
+   *
+   * @return true if the variable definition doesn't have usage clause
+   */
+  public boolean doesntHaveUsage() {
+    return usageClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition has redefines clause
+   *
+   * @return true if the variable definition has redefines clause
+   */
+  public boolean hasRedefines() {
+    return !redefinesClauses.isEmpty();
+  }
+
+  /**
+   * Return true if the variable definition doesn't have redefines clause
+   *
+   * @return true if the variable definition doesn't have redefines clause
+   */
+  public boolean doesntHaveRedefines() {
+    return redefinesClauses.isEmpty();
+  }
+
+  /**
+   * Return list of errors.
+   *
+   * @param variableNode related variableNode for extracting appropriate locality for errors
+   * @return the list of syntax errors
+   */
+  public List<SyntaxError> getErrors(VariableNode variableNode) {
+    return Stream.of(
+        checkClauseIsSingle(variableNode, this::getPicClauses, "PICTURE"),
+        checkClauseIsSingle(variableNode, this::getOccursClauses, "OCCURS"),
+        checkClauseIsSingle(variableNode, this::getValueClauses, "VALUE"),
+        checkClauseIsSingle(variableNode, this::getRedefinesClauses, "REDEFINES"),
+        checkClauseIsSingle(variableNode, this::getUsageClauses, "USAGE")
+        // TODO: check the same way that the other clauses are singular or absent
+    )
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  private static SyntaxError checkClauseIsSingle(VariableNode variableNode,
+                                                 Supplier<List> clausesGetter,
+                                                 String clauseName) {
+    if (clausesGetter.get().size() > 1)
+      return variableNode.getError(MessageTemplate.of(TOO_MANY_CLAUSES_MSG, clauseName));
+    return null;
+  }
+
+  /**
+   * Construct a new builder for {@link}VariableDefinitionNode{@link}.
+   *
+   * @return the builder
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * The builder for {@link}VariableDefinitionNode{@link}.
+   */
+  public static final class Builder {
```

Comment:
I didn't try it myself, but maybe `@Setter` and `@Accessors(fluent=true, chain=true)` will work

Reply:
Wow! It works!